roop/core.py

```diff
@@ -194,10 +195,14 @@ def start() -> None:
     extract_frames(roop.globals.target_path)
     temp_frame_paths = get_temp_frame_paths(roop.globals.target_path)
     update_status('Swapping in progress...')
-    conditional_process_video(roop.globals.source_path, temp_frame_paths)
-    # prevent memory leak using ffmpeg with cuda
+    conditional_process_video(roop.globals.source_path, temp_frame_paths, process_video)
     if roop.globals.gpu_vendor == 'nvidia':
         torch.cuda.empty_cache()
+    update_status('enhancinging in progress...')
+    conditional_process_video(roop.globals.source_path, temp_frame_paths, roop.enhancer.process_video)
+    # prevent memory leak using ffmpeg with cuda
```

Comment:
I guess this can go to a method release_resources() that you put right under limit_resources() and we call it after swapper and enhancer once